src/test/java/com/amazon/opendistroforelasticsearch/sql/correctness/report/FailedTestCase.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.correctness.report;
+
+import com.amazon.opendistroforelasticsearch.sql.correctness.runner.resultset.DBResult;
+import com.amazon.opendistroforelasticsearch.sql.correctness.runner.resultset.Row;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Report for test case that fails due to inconsistent result set.
+ */
+public class FailedTestCase extends TestCaseReport {
+
+    /** Inconsistent result set for reporting */
+    private final List<DBResult> results;
+
+    public FailedTestCase(String sql, List<DBResult> results) {
+        super(false, sql);
+        this.results = results;
+        this.results.sort(Comparator.comparing(DBResult::getDatabaseName));
+    }
+
+    @Override
+    public JSONObject report() {
```

Comment:
Is it possible to construct the report object and serialize the object to json. It may more clear to represent what is include in the report.

Reply:
Good point! Changed all report classes to generate JSON from object directly. And also use Lombok to simplify the code.